runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/JobInvocation.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.beam.runners.fnexecution.jobsubmission;
+
+import io.grpc.stub.StreamObserver;
+import org.apache.beam.model.jobmanagement.v1.JobApi.JobMessage;
+import org.apache.beam.model.jobmanagement.v1.JobApi.JobState;
+
+/**
+ * Internal representation of a Job which has been invoked (prepared and run) by a client.
+ */
+public interface JobInvocation {
+
+  /**
+   * Start the job.
+   */
+  void start();
+
+  /**
+   * @return Unique identifier for the job invocation.
+   */
+  String getId();
+
+  /**
+   * Cancel the job.
+   */
+  void cancel();
+
+  /**
+   * Retrieve the job's current state.
+   */
+  JobState.Enum getState();
+
+  /**
+   * Observe job state changes with a {@link StreamObserver}.
+   */
+  void addStateObserver(StreamObserver<JobState.Enum> stateStreamObserver);
```

Comment:
Consider defining an interface which is effectively a Consumer<JobState.Enum> instead of this being a StreamObserver.

Reply:
Can you expand on the motivation a little? By doing this, aren't we removing the possibility of the stream indicating an error?